### PR TITLE
1960 configuration repository

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Implements a concrete `IAgentConfigurationRepository`

Issue #1960

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/common/configuration/agent_configuration.py [hunk 1]
```diff
@@ -14,6 +14,224 @@
     PropagationConfiguration,
 )
 
+DEFAULT_AGENT_CONFIGURATION = """{
```
Comment (shreyamalviya) [inline monkey/common/configuration/agent_configuration.py:17]: Maybe move this to its own file, for readability.

File: monkey/monkey_island/cc/repository/file_agent_configuration_repository.py [hunk 2]
```diff
@@ -12,10 +20,15 @@ def __init__(self, file_repository: IFileRepository):
...
-
-        return self._schema.loads(configuration_json)
+        try:
+            with self._file_repository.open_file(AGENT_CONFIGURATION_FILE_NAME) as f:
+                configuration_json = f.read().decode()
+
+            return self._schema.loads(configuration_json)
+        # TODO: Handle FileRetrievalError vs FileNotFoundError
+        #       https://github.com/guardicore/monkey/blob/e8001d8cf76340e42bf17ff62523bd2d85fc4841/monkey/monkey_island/cc/repository/file_storage/local_storage_file_repository.py#L47-L50
+        except RetrievalError:
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repository/file_agent_configuration_repository.py:30]: Why not FileRetrievalError?

File: monkey/tests/monkey_island/single_file_repository.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import io
+from typing import BinaryIO
+
+from monkey_island.cc.repository import FileRetrievalError, IFileRepository
+
+
+class SingleFileRepository(IFileRepository):
```
[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/tests/monkey_island/single_file_repository.py:7]: Oops

File: monkey/monkey_island/cc/repository/file_agent_configuration_repository.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/repository/file_agent_configuration_repository.py:30]: As the TODO says, there's more work to be done here. The next PR will add a new error type and have it handle that instead.

**Review by shreyamalviya — APPROVED** [review]

**Pull request merged**